Add offset and number to create-guest-authors

aaronjorbin opened the pull request:
Currently, create-guest-authors is an all or nothing proposition. What happens though when you are 50% complete with a 30,000 user import and then accidently close your terminal window?  You have to start over.  By allowing chunking of create-guest-authors, this can be continued from the number that have been completed rather than starting from scratch.  

Anecdotally, running it in chunks seems to be quicker.

GaryJones commented:
Only the [last commit](https://github.com/Automattic/Co-Authors-Plus/pull/378/commits/f6bb3e2b062735ebef746d32be10ae12788e4393) is relevant here. Simple but useful addition.

aaronjorbin committed:
feat: add --offset and --number to create-guest-authors

Allow chunking of the create-guest-authors WP-CLI command so a long-running
import can be resumed rather than restarted. Previously, a 30,000-user
import that was interrupted partway through (for example, by an accidentally
closed terminal) had to be re-run from scratch. Passing --offset and
--number lets an operator pick up from where the previous run finished.

GaryJones committed:
docs: document --offset and --number on create-guest-authors

Make the chunking flags discoverable through `wp help` by replacing the
one-line subcommand summary with a description and an OPTIONS block, and
clarify in the docblock that the command is safe to re-run because users
with an existing guest author are skipped, so an interrupted import can be
resumed simply by invoking the command again. Add an inline comment beside
the `get_users()` call recording the stable-ordering assumption that
chunked runs depend on. Cover both behaviours with Behat scenarios so
future regressions surface in CI.
